### PR TITLE
Added support for bitbucket public repo

### PR DESCRIPTION
**Pull request by shrimpy**
This PR enables one click deploy from BitBucket, works for both HG and GIT public repo.

### CODE REVIEW

File: Slingshot.Api/Helpers/Constants.cs [hunk 1]
```diff
@@ -18,6 +13,19 @@ public class Repository
...
+            public const string BitBucketApiCommitsInfoFormat = "https://api.bitbucket.org/2.0/repositories/{0}/{1}/commits/{2}";
+
+            // BUG: "Missing default branch in v2 repository API when query public repo"
+            // https://bitbucket.org/site/master/issues/11732/missing-default-branch-in-v2-repository
+            //var defaultBranch = responseObj["default_branch"];
+            //if (defaultBranch == null)
+            //{
+            //    throw new ArgumentException("Could not discover default branch from repository");
+            //}
+            //_branch = defaultBranch.Value<string>();
```
Comment (ahmelsayed) [inline Slingshot.Api/Helpers/Constants.cs]: What is the commented code for and why is it in a Constants class?

Comment (shrimpy) [inline Slingshot.Api/Helpers/Constants.cs]: the comment is to remind us to use v2 API later. i will remove the commented code.

**Comment by davidebbo** [comment]
Do you have sample Bitbucket repo to test this with? e.g.

- with and without template
- with non-default branches

**Comment by shrimpy** [comment]
I tested with below two repo.
https://bitbucket.org/shrimpywu/publicgit
https://bitbucket.org/shrimpywu/publichg

they all have two branches and doesn`t have any template

**Pull request merged**